src/sdk/pynni/nni/evolution_tuner/evolution_tuner.py

```diff
@@ -115,20 +101,69 @@ def update_search_space(self, search_space):
 
         self.random_state = np.random.RandomState()
         self.population = []
-        is_rand = dict()
-
-        for item in self.space:
-            is_rand[item] = True
-
-        for _ in range(self.population_size):
-            config = json2parameter(
-                self.searchspace_json, is_rand, self.random_state)
-            self.population.append(Individual(config=config))
 
+    def trial_end(self, parameter_id, success, **kwargs):
+        """
+        To deal with trial failure. If a trial fails, readd the trial parameters into population.
```

Comment:
"readd"?

Reply:
I modify the docstring for clarity, please review on the latest version, thanks~